common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RetryingBlockTransferor.java

```diff
@@ -84,6 +87,12 @@ void createAndStart(String[] blockIds, BlockTransferListener listener)
   // while inside a synchronized block.
   /** Number of times we've attempted to retry so far. */
   private int retryCount = 0;
+  /**
+   * Map to track blockId to exception that the block is being retried for.
+   * This is mainly used in the case of SASL retries, because we need to set
+   * `retryCount` back to 0 in those cases.
+   */
+  private Map<String, Throwable> blockIdToException;
```

Comment:
I don't think this mapping is needed. You can reset  the `retryCount` to 0 after there aren't any more SASl timeouts to address @mridulm's comment. You may just need another flag that marks whether the last timeout was a SaslTimeout or not.

Reply:
yes, added this. this would be simpler.